### PR TITLE
cephfs-mirror: sanitize `daemons status` JSON

### PR DESCRIPTION
**Pull request by vshankar**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by vshankar** [comment]
jenkins test make check

**Comment by vshankar** [comment]
jenkins test make check

File: src/pybind/mgr/mirroring/fs/snapshot_mirror.py [hunk 1]
```diff
@@ -745,25 +745,39 @@ def daemon_status(self, filesystem):
...
-                                daemons[daemon_id] = json.loads(daemon_status['status_json'])
-                            except KeyError:
-                                # temporary, should get updated soon
-                                log.debug(f'daemon status not yet available for daemon_id {daemon_id}')
-                        except ValueError:
-                            pass
+                log.debug(f'daemon_entry: {daemon_entry}')
+                for daemon_key in daemon_entry.get('daemons', []):
+                    try:
+                        daemon_id = int(daemon_key)
```
Comment (leseb) [inline src/pybind/mgr/mirroring/fs/snapshot_mirror.py:754]: After thought, this is just the `daemon_name` right? It just happens to be an ID in this case.

Comment (vshankar) [reply src/pybind/mgr/mirroring/fs/snapshot_mirror.py:754]: right, the RADOS instance id.

**Comment by vshankar** [comment]
(update to fix tox failures -- needed type hints for that)

**Review by leseb — APPROVED** [review]

**Comment by vshankar** [comment]
https://pulpito.ceph.com/vshankar-2021-04-23_09:40:03-fs-wip-vshankar-testing-20210423-003830-distro-basic-gibba/

Failure is: https://tracker.ceph.com/issues/50224

**Pull request merged**